src/adapter/src/coord/command_handler.rs

```diff
@@ -372,43 +369,81 @@ impl Coordinator {
 
     /// Handles an execute command.
     #[tracing::instrument(level = "debug", skip_all)]
-    pub(crate) async fn handle_execute(&mut self, portal_name: String, mut ctx: ExecuteContext) {
-        if ctx.session().vars().emit_trace_id_notice() {
+    pub(crate) async fn handle_execute(
+        &mut self,
+        portal_name: String,
+        mut session: Session,
+        tx: ClientTransmitter<ExecuteResponse>,
+        // If this command was part of another execute command
+        // (for example, executing a `FETCH` statement causes an execute to be
+        //  issued for the cursor it references),
+        // then `outer_context` should be `Some`.
+        // This instructs the coordinator that the
+        // outer execute should be considered finished once the inner one is.
```

Comment:
Does this correctly handle things like:

```
BEGIN;
DECLARE c FOR VALUES (1), (2);
FETCH 1 c;
FETCH 1 c;
FETCH 1 c;
COMMIT;
```

Where the first `FETCH` only returns a partial result (even though it is fully executed)?

Also note this exact same sequence can happen by a driver, pgwire sending `Execute` messages with `max_rows`: https://github.com/MaterializeInc/materialize/blob/b636580062bbce260c04a665e613c6dbbdd7b624/test/pgtest/portals.pt#L8

Reply:
No, it panics. Good catch.